app/models/course/announcement/settings.rb

```diff
@@ -0,0 +1,40 @@
+class Course::Announcement::Settings
```

Comment:
I think should be `Course::Announcement::Settings` ? similar to `Course::Settings`